Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptions.swift

```diff
@@ -4,120 +4,114 @@
 
 import Foundation
 
+public typealias ChatMessageLayoutOptions = Set<ChatMessageLayoutOption>
+
+public extension ChatMessageLayoutOptions {
+    var rawValue: String {
+        map(\.rawValue).joined(separator: "-")
+    }
+
+    mutating func remove(_ options: ChatMessageLayoutOptions) {
+        self = subtracting(options)
+    }
+
+    mutating func insert(_ options: ChatMessageLayoutOptions) {
+        options.forEach { self.insert($0) }
+    }
+
+    /// Returns all options the current option set consists of separated by `-` character.
+    var description: String {
+        Self.singleOptions
+            .compactMap { contains($0) ? $0.optionName : nil }
+            .joined(separator: "-")
+    }
+
+    static let singleOptions: [ChatMessageLayoutOption] = [
+        .flipped,
+        .bubble,
+        .continuousBubble,
+        .avatarSizePadding,
+        .avatar,
+        .timestamp,
+        .authorName,
+        .text,
+        .quotedMessage,
+        .threadInfo,
+        .errorIndicator,
+        .reactions,
+        .onlyVisibleForYouIndicator,
+        .centered
+    ]
```

Comment:
@evsaev I did not understand why we need these? 🤔

Reply:
+1. What does singleOptions mean? is it meant to be like the `allCases` in an enum?